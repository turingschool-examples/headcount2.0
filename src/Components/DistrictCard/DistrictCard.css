.highStats {
  color: #7097B2;
}

.lowStats {
  color: #B81F38;
}

.card {
  border-radius: 18px;
  border: solid #4F4C47 2px;
  color: #4F4C47;
  display: inline-block;
  margin: 1rem;
  padding: .5rem;
  width: 300px;
}

.selected {
  background-color: #F7F5F1;
  border: solid #21201E 4px;
}

.card:hover {
  border: solid #4F4C47 4px;
}

.district {
  font-family: avenir, sans-serif;
  font-size: 1.5em;
  text-align: center;
}

.stats {
  font-family: avenir, sans-serif;
  font-size: 1.25em;
  padding-left: 5rem;
}