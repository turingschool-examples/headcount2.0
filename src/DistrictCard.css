.districtCard {
  box-shadow: 1em;
  border-radius: 15px;
  background-color: white;
  display: flex;
  flex-direction: column;
  width: 15rem;
  height: auto;
  margin: 0 0 5rem 0;
  padding-bottom: 1rem;
  text-align: center
}

.districtLocation {
  font-size: 1rem
}

.districtStats {
  list-style-type: none;
  text-align: center;
  margin: 0;
  padding: 0
}

.statGreater {
  color: blue
}

.statLesser {
  color: red
}