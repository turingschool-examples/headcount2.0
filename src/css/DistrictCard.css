.below-five {
  color: forestgreen;
}

.above-five {
  color: slateblue;
}

.each-card {
  border: 4px solid blanchedalmond;
  border-radius: 8px;
  margin: 0.75rem;
  width: 15rem;
  background: aquamarine;
  padding-bottom: 1rem;
  -webkit-box-shadow: 1px 1px 7px 1px black;
  box-shadow: 1px 1px 7px 1px black;
}

.outline {
  border: 4px solid black;
}

h1 {
  font-size: 1rem;
}

section {
  font-size: 0.8rem;
}

section,
h1 {
  text-align: center;
}
