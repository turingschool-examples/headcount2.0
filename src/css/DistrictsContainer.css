.district-container {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  justify-items: center !important;
}
