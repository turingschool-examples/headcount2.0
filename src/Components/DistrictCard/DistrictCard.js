import React from 'react';
import './DistrictCard.css';
import PropTypes from 'prop-types';


const DistrictCard = ({district , selectDistrict}) => {
  const years = Object.keys(district.stats)

  const determineColor = (stats) => {
    return stats > 0.5 ? "highStats" : "lowStats"
  }

  const createStats = years.map((year, index) => {
    const stats = district.stats[year]
    return (
      <li className={determineColor(stats)} key={index}>
        {[year]}: {stats}
      </li>
      )
  })

  const handleClick = (location) => {
    selectDistrict(location)
  }

  return (
    <section 
      className={`card ${district.selected ? 'selected' : ''}`} 
      id={district.key}
      key={district.key}
      onClick={() => {handleClick(district.location)}}
    > 
      <h2 className="district">District Name: {district.location}</h2>
      <ul className="stats">
        {createStats}
      </ul>
    </section>
  )
}

DistrictCard.propTypes = { 
  district: PropTypes.object
}



export default DistrictCard;