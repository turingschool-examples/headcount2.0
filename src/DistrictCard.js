import React from "react";
import "./CSS/DistrictCard.css";

export const DistrictCard = ({ location, stats, selectCard }) => {
  return (
    <div className="district-card">
      <h1 className="district-location">{location}</h1>
      {Object.keys(stats).map((year, i) => {
        const avg = stats[year];
        return (
          <section
            className={avg < 0.5 ? "year-red" : "year-black"}
            key={i}
            onClick={e => selectCard(location)}
          >
            {year} :
            <span className={avg < 0.5 ? "year-red" : "year-black"}>{avg}</span>
          </section>
        );
      })}
    </div>
  );
};
