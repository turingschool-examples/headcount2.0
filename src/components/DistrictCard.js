import React from "react";
import "../css/DistrictCard.css";
import PropTypes from "prop-types";

const DistrictCard = ({ location, stats, selectCards, selected }) => {
  const districtYears = Object.keys(stats).map((year, index) => {
    const avg = stats[year];
    return (
      <article
        className={avg < 0.5 ? "below-five" : "above-five"}
        key={index}
        onClick={() => selectCards(location)}
      >
        {[year]}: {avg}
      </article>
    );
  });
  return (
    <div className={`each-card ${selected ? "outline" : "each-card"}`}>
      <h1>{location}</h1>
      <section>{districtYears}</section>
    </div>
  );
};

DistrictCard.propTypes = {
  location: PropTypes.string,
  stats: PropTypes.object,
  selectCards: PropTypes.func,
  selected: PropTypes.bool
};

export default DistrictCard;
