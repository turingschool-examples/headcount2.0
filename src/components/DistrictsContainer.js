import React from "react";
import DistrictCard from "./DistrictCard";
import "../css/DistrictsContainer.css";
import PropTypes from "prop-types";

const DistrictsContainer = ({ districts, selectCards }) => {
  const districtCard = districts.map((district, index) => (
    <DistrictCard {...district} key={index} selectCards={selectCards} />
  ));
  return <div className="district-container">{districtCard}</div>;
};

DistrictsContainer.propTypes = {
  districts: PropTypes.array,
  selectCards: PropTypes.func
};

export default DistrictsContainer;
