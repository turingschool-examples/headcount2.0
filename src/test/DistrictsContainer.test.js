import React from "react";
import { shallow } from "enzyme";
import DistrictsContainer from "../components/DistrictsContainer";
import DistrictCard from "../components/DistrictCard";

describe("DistrictsContainer", () => {
  const mockCard = [
    { "FEE Room G": { 2018: 19 }, "FEE Room E": { 2018: 19 } },
    { "BEE Room G": { 2018: 14 }, "BEE Room E": { 2018: 12 } }
  ];
  let wrapper;
  let anticipated;

  it("should match the snapshot", () => {
    wrapper = shallow(<DistrictsContainer districts={mockCard} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render all found districts", () => {
    wrapper = shallow(<DistrictsContainer districts={mockCard} />);
    anticipated = wrapper.find(DistrictCard).length;
    expect(anticipated).toEqual(2);
  });
});
