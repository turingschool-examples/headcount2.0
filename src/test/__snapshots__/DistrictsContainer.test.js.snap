// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictsContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictsContainer
    districts={
        Array [
            Object {
              "FEE Room E": Object {
                "2018": 19,
              },
              "FEE Room G": Object {
                "2018": 19,
              },
            },
            Object {
              "BEE Room E": Object {
                "2018": 12,
              },
              "BEE Room G": Object {
                "2018": 14,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DistrictCard
          FEE Room E={
                    Object {
                              "2018": 19,
                            }
          }
          FEE Room G={
                    Object {
                              "2018": 19,
                            }
          }
          selectCards={undefined}
/>,
        <DistrictCard
          BEE Room E={
                    Object {
                              "2018": 12,
                            }
          }
          BEE Room G={
                    Object {
                              "2018": 14,
                            }
          }
          selectCards={undefined}
/>,
      ],
      "className": "district-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "FEE Room E": Object {
            "2018": 19,
          },
          "FEE Room G": Object {
            "2018": 19,
          },
          "selectCards": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "BEE Room E": Object {
            "2018": 12,
          },
          "BEE Room G": Object {
            "2018": 14,
          },
          "selectCards": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DistrictCard
            FEE Room E={
                        Object {
                                    "2018": 19,
                                  }
            }
            FEE Room G={
                        Object {
                                    "2018": 19,
                                  }
            }
            selectCards={undefined}
/>,
          <DistrictCard
            BEE Room E={
                        Object {
                                    "2018": 12,
                                  }
            }
            BEE Room G={
                        Object {
                                    "2018": 14,
                                  }
            }
            selectCards={undefined}
/>,
        ],
        "className": "district-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "FEE Room E": Object {
              "2018": 19,
            },
            "FEE Room G": Object {
              "2018": 19,
            },
            "selectCards": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "BEE Room E": Object {
              "2018": 12,
            },
            "BEE Room G": Object {
              "2018": 14,
            },
            "selectCards": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
